packages/gotrue/lib/src/gotrue_client.dart

```diff
@@ -951,7 +951,7 @@ class GoTrueClient {
       notifyAllSubscribers(AuthChangeEvent.tokenRefreshed);
       _refreshTokenCompleter!.complete(authResponse);
       return authResponse;
-    } on SocketException {
```

Comment:
I'm a bit confused about what is going on here. How can the thrown error type change without changing anything within the try block? Wouldn't `.request()` call on line 942 still throw SocketException when there is an issue with the network?

Reply:
From the [doc](https://pub.dev/documentation/http/latest/io_client/IOClient-class.html)
> If there is a socket-level failure when communicating with the server (for example, if the server could not be reached), [IOClient](https://pub.dev/documentation/http/latest/io_client/IOClient-class.html) will emit a [ClientException](https://pub.dev/documentation/http/latest/http/ClientException-class.html) that also implements [SocketException](https://api.dart.dev/stable/3.1.2/dart-io/SocketException-class.html). This allows callers to get more detailed exception information for socket-level failures, if desired.

So any SocketException from the HttpClient gets caught and put into a ClientException that also implements SocketException, so you can use both to catch the exception. See their [send implementation](https://pub.dev/documentation/http/latest/io_client/IOClient/send.html).